api/v1alpha1/hosted_controlplane.go

```diff
@@ -58,6 +58,10 @@ type HostedControlPlaneSpec struct {
 	// in the secret with a secret key name that corresponds to the constant AuditWebhookKubeconfigKey.
 	// +optional
 	AuditWebhook *corev1.LocalObjectReference `json:"auditWebhook,omitempty"`
+
+	// Etcd contains metadata about the etcd cluster the hypershift managed Openshift control plane components
+	// uses to store data.
+	Etcd EtcdSpec `json:"etcd"`
```

Comment:
@csrwng @enxebre @sjenning @relyt0925 

Details I think we should sort out from this point downward before merging:

1. Docs on fields
2. API metadata tagging for optionality, enum types, etc
3. Defaulting

Reply:
And something else I just thought of which is significant: what parts of this field are immutable? If not, what happens if the type is mutated? It needs clarified/documented (and if immutable, enforced somehow).

Reply:
the top makes sense but your immutable comment I would need more clarity into the strategy before doing

Reply:
For now maybe just update the doc to mention that changing the management type is unsupported (even though it won't be enforced yet)?

Reply:
done deal